ang/civicase/MasonryGrid.js

```diff
@@ -0,0 +1,118 @@
+      (function init () {
+        $timeout(function () {
+          appendMasonryContainers();
+          arrangeGridItems();
+          $scope.$on('civicaseMasonryGrid::updated', arrangeGridItems);
+        });
+      })();
+
+      /**
+       * Appends the masonry containers used for splitting the grid items.
+       */
+      function appendMasonryContainers () {
+        _.times(2).forEach(function () {
+          $('<div></div>')
+            .addClass('civicase__summary-tab-tile')
+            .addClass('civicase__masonry-grid__container')
+            .appendTo($element);
+        });
+      }
+
+      /**
+       * Sorts the grid items into one of the two grid containers.
+       */
+      function arrangeGridItems () {
+        $timeout(function () {
+          $element.find('[civicase-masonry-grid-item]').detach();
+
+          masonryGrid.$gridItems.forEach(function ($gridItem, index) {
+            var columnIndex = index % 2;
+            var $column = $element.find('.civicase__masonry-grid__container').eq(columnIndex);
+
+            $gridItem.appendTo($column);
+          });
+        });
+      }
+    }
+  });
+
+  module.controller('civicaseMasonryGridController', function ($scope) {
+    var vm = this;
+    vm.$gridItems = [];
+
+    /**
+     * Adds an item element to the grid.
+     *
+     * @param {Object} $gridItem a reference to the item element to be added.
+     */
+    vm.addGridItem = function ($gridItem) {
+      _.remove(vm.$gridItems, $gridItem);
+      vm.$gridItems.push($gridItem);
+      $scope.$emit('civicaseMasonryGrid::updated');
+    };
+
+    /**
+     * Adds an item element to the grid at the given index.
+     *
+     * @param {Object} $gridItem a reference to the item element to be added.
+     */
+    vm.addGridItemAt = function ($gridItem, atIndex) {
+      _.remove(vm.$gridItems, $gridItem);
+      vm.$gridItems.splice(atIndex, 0, $gridItem);
+      $scope.$emit('civicaseMasonryGrid::updated');
+    };
+
+    /**
+     * Removes the item element from the grid.
+     *
+     * @param {Object} $gridItem a reference to the item element to be removed.
+     */
+    vm.removeGridItem = function ($gridItem) {
+      _.remove(vm.$gridItems, $gridItem);
+      $scope.$emit('civicaseMasonryGrid::updated');
+    };
+  });
+
+  module.directive('civicaseMasonryGridItem', function () {
+    return {
+      require: '^civicaseMasonryGrid',
+      link: civicaseMasonryGridItemLink
+    };
+
+    /**
+     * Masonry grid item link function.
+     *
+     * @param {Object} $scope the directive's scope.
+     * @param {Object} $element a reference to the directive's element.
+     * @param {Object} attrs a map of attributes associated to the element.
+     * @param {Object} masonryGrid a reference to the parent masonry grid directive's controller.
+     */
+    function civicaseMasonryGridItemLink ($scope, $element, attrs, masonryGrid) {
+      (function init () {
+        masonryGrid.addGridItem($element);
```

Comment:
the logic seems to be in the wrong place...it would make more sense to have this done by `civicaseMasonryGrid`: when the directive initializes, it finds inside itself all the elements marked as `civicase-masonry-grid-item` and  adds them to its controller's list

This directive doesn't really do anything other than doing a job that the parent directive should do, it would be better if you removed it

Reply:
This is how AngularJS recommends child-parent interactions:
https://docs.angularjs.org/guide/directive#creating-directives-that-communicate

Also, when the child has control over when to append itself to the parent, it allows more dynamic control of when to show or hide the element from parent grid.